deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/worker/DeployTagWorker.java

```diff
@@ -122,14 +124,17 @@ private void processBatch() throws Exception {
                 LOG.info("process job: {}", job);
                 String lockName = String.format("DeployTagWorker-%s", job.getConstraint_id());
                 Connection connection = utilDAO.getLock(lockName);
+                DeployConstraintBean latestJob = deployConstraintDAO.getById(job.getConstraint_id());
                 if (connection != null) {
                     try {
-                        processEachEnvironConstraint(job);
+                        processEachEnvironConstraint(latestJob);
                     } catch (Exception e) {
-                        LOG.error("failed to process job: {}", job.toString(), e);
-                        job.setState(TagSyncState.ERROR);
-                        deployConstraintDAO.updateById(job.getConstraint_id(), job);
-                        LOG.error("updated job state to {}", TagSyncState.ERROR);
+                        LOG.error("failed to process job: {} Error {} stack {}", latestJob.toString(), ExceptionUtils.getRootCauseMessage(e), ExceptionUtils.getStackTrace(e));
+                        if (!SQLException.class.isInstance(e)) {
+                            latestJob.setState(TagSyncState.ERROR);
+                            deployConstraintDAO.updateById(job.getConstraint_id(), latestJob);
```

Comment:
we use job and latestJob here. maybe they should be the same?

Reply:
https://github.com/pinterest/teletraan/pull/705#discussion_r394589964